src/modules/DNS.rb

```diff
@@ -246,24 +246,16 @@ def default_dhcp_hostname
     end
 
     def ReadHostname
-      fqhostname = ""
       # In installation (standard, or AutoYaST one), prefer /etc/install.inf
       # (because HOSTNAME comes with netcfg.rpm already, #144687)
       if (Mode.installation || Mode.autoinst) && FileUtils.Exists("/etc/install.inf")
         fqhostname = read_hostname_from_install_inf
       end
 
-      fqhostname = read_hostname_from_etc if fqhostname.empty?
+      # reads setup from /etc/HOSTNAME, returns a default if nothing found
+      fqhostname = Hostname.CurrentFQ if fqhostname.nil? || fqhostname.empty?
 
-      split = Hostname.SplitFQ(fqhostname)
-      @hostname = split[0] || ""
-      @domain = split[1] || ""
-
-      # last resort
-      if @hostname == ""
-        @hostname = "linux"
-        @domain = "site"
-      end
+      @hostname, @domain = *Hostname.SplitFQ(fqhostname)
```

Comment:
SplitFQ may return `[foo, bar]` or `[foo]`. Are you sure you want a nil in `@domain`?